### PR TITLE
Replaced Mapbox with osmdroid (Nearby activity)

### LINKED ISSUE #5372: Replace Mapbox with osmdroid (Nearby activity)
The Nearby activity shows a map using the Mapbox library.

We [need](https://github.com/commons-app/apps-android-commons/issues/5352) to switch to the **osmdroid** library:
https://github.com/osmdroid/osmdroid/wiki/How-to-use-the-osmdroid-library-(Kotlin)

The task here is to perform this switch, with the following feature still working:
- Map gets displayed, centered on the user's location
- Map can be moved/zoomed with the fingers (osmdroid default behaviors are fine)
- A pin appear on the map for each Wikidata item
- Tapping on a pin opens the bottom UI with +/name/etc
- The other features _(such as place state/type, dark mode, user position, locale)_ are not really required for now, they will be the topic of future other GitHub issues.

Any tiles server that works can be used for now.

### PR DESCRIPTION
**Pull request by kanahia1**
Fixes #5372 

What changes did you make and why?
✔️ Replaced MapBox with OSMDroid
✔️ Tested most of the pre-existing features

**Tests performed (required)**

Tested betaDebug on Samsung S21 FE with API level 33.

**Screenshots (for UI changes only)**
https://github.com/commons-app/apps-android-commons/assets/114223204/501e79a8-bf8e-4a3a-8499-f0490e46af3c

🔧 Need to work on tests

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java [hunk 1]
```diff
@@ -357,71 +396,126 @@ public void onViewCreated(@NonNull final View view, @Nullable final Bundle saved
...
+        mMapView.addMapListener(new MapListener() {
+            @Override
+            public boolean onScroll(ScrollEvent event) {
+                if (lastMapFocus != null) {
+                    Location mylocation = new Location("");
+                    Location dest_location = new Location("");
+                    dest_location.setLatitude(mMapView.getMapCenter().getLatitude());
+                    dest_location.setLongitude(mMapView.getMapCenter().getLongitude());
+                    mylocation.setLatitude(lastMapFocus.getLatitude());
+                    mylocation.setLongitude(lastMapFocus.getLongitude());
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java:472]: I killed+restarted the app many times, but it still shows me (the blue dot) at the place where I was 4 hours ago.
I wonder if something here may need to be updated?
The app has always been a bit slow to update the location, but killing+restarting the app always fixed it.

File: app/src/main/java/fr/free/nrw/commons/nearby/RotationGestureOverlay.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import org.osmdroid.views.overlay.Overlay;
+import org.osmdroid.views.overlay.gestures.RotationGestureDetector;
+
+public class RotationGestureOverlay extends Overlay implements
+    RotationGestureDetector.RotationListener, IOverlayMenuProvider {
+    private final static boolean SHOW_ROTATE_MENU_ITEMS = false;
+
+    private final static int MENU_ENABLED = getSafeMenuId();
+    private final static int MENU_ROTATE_CCW = getSafeMenuId();
+    private final static int MENU_ROTATE_CW = getSafeMenuId();
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/nearby/RotationGestureOverlay.java:24]: Sorry to not have conveyed this earlier: Map rotation is not really needed. Feel free to not remove it, especially if rotation makes the code more complex. Tiles seem not often not update after rotation, which impacts readability, so no rotation might be better.

**Review by nicolas-raoul — COMMENTED** [review]
It works great!
Hopefully unit tests can be fixed. :-)
I just added a few comments.
Thanks a lot for the nice code!

File: app/src/main/java/fr/free/nrw/commons/nearby/RotationGestureOverlay.java (hunk 2 above)
Comment (kanahia1) [reply app/src/main/java/fr/free/nrw/commons/nearby/RotationGestureOverlay.java]: Oh no worries, I have removed rotation feature in newer commit

File: app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java (hunk 1 above)
Comment (kanahia1) [reply app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java:444]: I have fixed this issue and tested it using android emulator in newer commit
https://drive.google.com/file/d/1ylx3JjyFRUlAS1pRmzn-Kypgz7eFKFvb/view?usp=sharing

File: app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java [hunk 3]
```diff
@@ -357,71 +396,126 @@ public void onViewCreated(@NonNull final View view, @Nullable final Bundle saved
...
+        mMapView.getZoomController().setVisibility(CustomZoomButtonsController.Visibility.NEVER);
+        mMapView.getController().setZoom(ZOOM_LEVEL);
+        MapTileProviderBasic tileProvider = new MapTileProviderBasic(this.getContext());
+        ITileSource tileSource = new XYTileSource(
+            "FietsRegionaal",
+            3,
+            18,
+            256,
+            ".png",
+            new String[]{"http://overlay.openstreetmap.nl/openfietskaart-rcn/"}
```
[the remark this answers is not in the document]
Comment (kanahia1) [reply app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java]: I have removed it in newer commit

File: app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java [hunk 4]
```diff
@@ -134,49 +122,103 @@
...
+    @BindView(R.id.search_view)
+    SearchView searchView;
+    @BindView(R.id.search_list_view)
+    RecyclerView recyclerView;
+    @BindView(R.id.nearby_filter_list)
+    View nearbyFilterList;
+    @BindView(R.id.checkbox_tri_states)
+    CheckBoxTriStates checkBoxTriStates;
+    @BindView(R.id.map)
+    org.osmdroid.views.MapView mMapView;
```
[the remark this answers is not in the document]
Comment (kanahia1) [reply app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java]: Sure, I have changed the `mMapView` to `mapView`

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
The location is now updated at least when restarting the app, thanks for fixing! 🙂

The unit tests are the next priority, I guess.

Minor thing: Would it be possible to close the pin labels at some point, for instance when tapping another pin, so that at most one label is open at any time? If difficult it can be done later, possibly by someone else, no worries.

**Comment by nicolas-raoul** [comment]
Feel free to add `@Ignore` to the tests that are failing, they can be solved later by anyone as another GitHub issue.
And pull from `main` to resolve the small conflict in `app/build.gradle`.
Then this pull request can be merged into the main branch. :-)

File: app/src/main/res/drawable/compass_arrow.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (kanahia1) [reply app/src/main/res/drawable/compass_arrow.xml]: I have removed them in newer commit and I have also fixed the marker click problem [`3a6365f` (#5403)](https://github.com/commons-app/apps-android-commons/pull/5403/commits/3a6365f54552f79ef59cdeeefa32da8960a1f7e1)

**Comment by kanahia1** [comment]
Hi @nicolas-raoul , I have made the changes as advised by you and I have also fixed the marker click issue. 

https://github.com/commons-app/apps-android-commons/assets/114223204/f69676a2-d648-4961-bf28-5cf387ac6c42

**Review by nicolas-raoul — APPROVED** [review]
Thanks a lot @kanahia1 for this very needed change!
It works great.

**Comment by nicolas-raoul** [comment]
I created https://github.com/commons-app/apps-android-commons/issues/5408

**Pull request merged**